addons/project/tests/test_project_recurrence.py

```diff
@@ -601,19 +612,29 @@ def get_task_and_subtask_counts(domain):
                 subtask_form.repeat_type = 'after'
                 subtask_form.repeat_number = 4
                 subtask_form.date_deadline = datetime(2020, 2, 3)
+                subtask_form.save()
 
+            grand_child_task_3 = self.env['project.task'].create({
+                'name': 'Grandchild task 3',
+            })
+            grand_child_task_4 = self.env['project.task'].create({
+                'name': 'Grandchild task 4',
+            })
+            grand_child_task_5 = self.env['project.task'].create({
+                'name': 'Grandchild task 5',
+            })
             # create non-recurring grandchild subtasks
             with Form(non_recurrent_subsubtask.with_context(tracking_disable=True)) as subtask_form:
-                with subtask_form.child_ids.new() as subsubtask_form:
-                    subsubtask_form.name = 'Grandchild task 3'
+                subtask_form.child_ids.add(grand_child_task_3)
+                subtask_form.save()
```

Comment:
Normally the save of the form object is automatically done after the `with` statement. 🙂 
```suggestion
```